src/layout/Layout.astro

```diff
@@ -0,0 +1,74 @@
+---
+//Styles
+import "@styles/site.css"
+
+//Assets
+import logo from "@images/logo.png"
+
+export interface Props {
+	pageTitle: string,
+	pageUrl: string,
+	pageImage: string,
+	seoTitle?: string,
+	pageDescription?: string,
+	bodyClassName?: string,
+	hideNav?: boolean
+}
+
+const {
+		pageTitle,
+		seoTitle = pageTitle, // This prop is used for when the title tag title is different from what is displayed for SEO titles
+		pageUrl,
+		pageImage,
+		pageDescription = "A collection of publishing experiments",
+		bodyClassName = "",
+		hideNav = false
+	} = Astro.props;
+---
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>{pageTitle}</title>
+
+		<slot name="header-content"/>
+		<meta name="title" content={seoTitle}>
+		<meta name="description" content={pageDescription}>
+        <meta name="viewport" content="width=device-width,initial-scale=1.0">
+
+		<!-- Open Graph / Facebook -->
+		<meta property="og:type" content="website">
+		<meta property="og:url" content={pageUrl}>
+		<meta property="og:title" content={seoTitle}>
+		<meta property="og:description" content={pageDescription}>
+		<meta property="og:image" content={pageImage}>
+
+		<!-- Twitter -->
+		<meta property="twitter:card" content="summary_large_image">
+		<meta property="twitter:url" content={pageUrl}>
+		<meta property="twitter:title" content={seoTitle}>
+		<meta property="twitter:description" content={pageDescription}>
+		<meta property="twitter:image" content={pageImage}>
+
+		<link rel="icon" type="image/x-icon" href="favicon.ico">
+	</head>
+    <body class={bodyClassName}>
+		{
+			!hideNav && (
+				<nav>
+					<div class="logo">
+						<a href="/">
+						<img src={logo} width="60px" height="60px" alt="Logo">
+						</a>
+					</div>
```

Comment:
@Oluwa-nifemi what's the use case here for optionally hiding the nav here. Do you mind showing screenshots or a video maybe

Reply:
Hey @Odusanya 
I used it for four pages
- The odabokayode page. It doesn't have the nav in the normal design
- The redirect pages (/redirect, /covers, /jguide). This is so that the nav doesn't flash on the page while redirecting them to their intended destination. I noticed it was happening while testing so I just decided to hide the nav totally so that the user will see the white page for the 1 or 2 seconds they're waiting to be redirected